### PR TITLE
Refresh Availabilities Feature

### LINKED ISSUE #60: Auto Remove availabilities feature
Once an availability is over, the date would be automatically removed from the app

### PR DESCRIPTION
**Pull request by tayziyi**
Fix #60 

Added a command to refresh outdated availabilities based on current date

Example Usage:
refresh

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/RefreshCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public RefreshCommand(Availability dateToDelete) {
+        requireNonNull(dateToDelete);
+        this.dateToDelete = dateToDelete;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Set<Person> personsToUpdate = model.getAddressBook().getPersonList().stream()
```
Comment (SimKianSeng) [inline src/main/java/seedu/address/logic/commands/RefreshCommand.java:42]: Possible deep nesting/arrowhead code at this section?

**Review by SimKianSeng — COMMENTED** [review]
Great job 👍🏻 ! The concept is there, but could this functionality be executed automatically by comparing with today's date to better fit our user story? Perhaps we can refer to how addressbook.json file is being updated periodically without a specific command for it?

If it is difficult to implement is it possible we park this feature as an extension to `removeavail` command? E.g. `removeavail before/01/01/2024` removes all availabilities before 1st Jan 2024.

Might want to consult the others @Benson15912 @xuwen-ho @YowSiaoKang for inputs.

**Comment by YowSiaoKang** [comment]
> Great job 👍🏻 ! The concept is there, but could this functionality be executed automatically by comparing with today's date to better fit our user story? Perhaps we can refer to how addressbook.json file is being updated periodically without a specific command for it?
> 
> If it is difficult to implement is it possible we park this feature as an extension to `removeavail` command? E.g. `removeavail before/01/01/2024` removes all availabilities before 1st Jan 2024.
> 
> Might want to consult the others @Benson15912 @xuwen-ho @YowSiaoKang for inputs.

Agree with Kian Seng that refresh should be automatic and removes all availability before system time.

**Comment by Benson15912** [comment]
I think maybe we dont make it a command but like a user setting i guess? Like turn on auto delete or not. Can probably discuss this during tomorrow's meeting. Functionality wise, i think we have to be careful with the dependency here with assignments.

**Review by SimKianSeng — APPROVED** [review]
LGTM

**Pull request merged**